resource/geo-density-chart.js

```diff
@@ -84,23 +84,31 @@ function GeoDensityChart(){
         // iterate in the layers of the map
         _mapPointer.eachLayer(function(layer){
             if (layer.options.id){
+                console.log("[MapTile] Received previous maptile id:", layer.options.id);
                 let url;
-                _mapPointer.removeLayer(layer);
+                let idTileMap;
+                console.log("[MapTile] Received asked maptile id", mapTile);
                 switch (mapTile) {
                     case 'cartodb-positron':
-                        url = 'http://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png';
+                        url = 'http://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}@2x.png';
+                        idTileMap = "light_all";
                         break;
                     case 'cartodb-dark':
-                        url = 'http://{s}.basemaps.cartocdn.com/dark_all/{z}/{x}/{y}.png';
+                        url = 'http://{s}.basemaps.cartocdn.com/dark_all/{z}/{x}/{y}@2x.png';
+                        idTileMap = "dark_all";
                         break;
                 }
-                L.tileLayer(url, {
-                    maxZoom: 18,
-                    attribution: 'Map data &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, &copy; <a href="http://cartodb.com/attributions">CartoDB</a>',
-                    id: 'light_all',
-                    tileSize: 512,
-                    zoomOffset: -1
-                }).addTo(_mapPointer).set;
+                if (layer.options.id !== idTileMap){
```

Comment:
To make it easier to understand, I suggest to name the boolean `layer.options.id !== idTileMap`
Perhaps : `isBackgroundUpdated` or `hasBackgroundChanged`or `backgroundChange`

Reply:
Done in d2c7118, add precision on the two compared variable